File: modernized/app/models/account.py
from sqlalchemy import BigInteger, Column, Numeric, String

from app.database import Base


class Account(Base):
    __tablename__ = "accounts"

    acct_id = Column(BigInteger, primary_key=True)
    active_status = Column(String(1))
    curr_bal = Column(Numeric(12, 2), default=0)
    credit_limit = Column(Numeric(12, 2), default=0)
    cash_credit_limit = Column(Numeric(12, 2), default=0)
    open_date = Column(String(10))
    expiration_date = Column(String(10))
    reissue_date = Column(String(10))
    curr_cyc_credit = Column(Numeric(12, 2), default=0)
    curr_cyc_debit = Column(Numeric(12, 2), default=0)
    addr_zip = Column(String(10))
    group_id = Column(String(10))

File: etl/src/main/java/com/carddemo/etl/model/Account.java
package com.carddemo.etl.model;

import java.math.BigDecimal;
import java.time.LocalDate;

/**
 * Modern representation of a CardDemo Account record (copybook {@code CVACT01Y} / VSAM
 * {@code ACCTDAT}). Field order and semantics mirror {@code ACCOUNT-RECORD}.
 *
 * @param acctId          {@code ACCT-ID} — PIC 9(11), primary key
 * @param activeStatus    {@code ACCT-ACTIVE-STATUS} — PIC X(01), 'Y' or 'N'
 * @param currBal         {@code ACCT-CURR-BAL} — PIC S9(10)V99
 * @param creditLimit     {@code ACCT-CREDIT-LIMIT} — PIC S9(10)V99
 * @param cashCreditLimit {@code ACCT-CASH-CREDIT-LIMIT} — PIC S9(10)V99
 * @param openDate        {@code ACCT-OPEN-DATE} — PIC X(10), ISO date or null
 * @param expirationDate  {@code ACCT-EXPIRAION-DATE} — PIC X(10), ISO date or null
 * @param reissueDate     {@code ACCT-REISSUE-DATE} — PIC X(10), ISO date or null
 * @param currCycCredit   {@code ACCT-CURR-CYC-CREDIT} — PIC S9(10)V99
 * @param currCycDebit    {@code ACCT-CURR-CYC-DEBIT} — PIC S9(10)V99
 * @param addrZip         {@code ACCT-ADDR-ZIP} — PIC X(10), trimmed or null
 * @param groupId         {@code ACCT-GROUP-ID} — PIC X(10), trimmed or null
 */
public record Account(
        long acctId,
        String activeStatus,
        BigDecimal currBal,
        BigDecimal creditLimit,
        BigDecimal cashCreditLimit,
        LocalDate openDate,
        LocalDate expirationDate,
        LocalDate reissueDate,
        BigDecimal currCycCredit,
        BigDecimal currCycDebit,
        String addrZip,
        String groupId) {
}
